frontend/app/player/MessageDistributor/managers/ListWalker.ts

```diff
@@ -118,4 +118,19 @@ export default class ListWalker<T extends Timed> {
 		}
 	}
 
+	async moveWait(t: number, fn: (msg: T) => Promise<any> | undefined, fnBack?: (msg: T) => void): Promise<void> {
+		// Applying only in increment order for now
+		if (t < this.timeNow) {
+			this.reset();
+		}
+
+		while (!!this._list[this.p] && this._list[this.p].time <= t) {
+			const ret = fn(this._list[ this.p++ ]);
+			if (ret) { await ret }
+		}
+		while (fnBack && this.p > 0 && this._list[ this.p - 1 ].time > t) {
```

Comment:
pretty much same here; good idea to make better name than `p`

Reply:
otherwise it looks like you're writing self obfuscating code

Reply:
What is not clear exactly? This is the ListWalker class with the pointer  `p` to the current element, there is no complex logic in it. It will be even more wordy to write "this.pointer" each time. Do you write `for` cycles like the following  `for (var iteration = 0; iteration < n; iteration++)` each time? I don't think so.

Reply:
Besides, this is a copy-paste hack of a function above for a single-use. I'll rethink that during player refactoring 2d phase.

Reply:
I made a list const to decrease amount of `this`